### PR TITLE
Docs: Documents caching encryption in enterprise config

### PR DESCRIPTION
**Pull request by stephaniehingtgen**
**What this PR does / why we need it**:
Documents the two new configuration flags in the enterprise-configuration docs used for encrypting the cache.

**Which issue(s) this PR fixes**:

n/a (internal)

**Special notes for your reviewer**:

### CODE REVIEW

File: docs/sources/enterprise/enterprise-configuration.md [hunk 1]
```diff
@@ -361,6 +361,19 @@ This value limits the size of a single cache value. If a cache value (or query r
...
+## [caching.encryption]
+### enabled
+
+Setting 'enabled' to `true` allows users to encrypt the query values in the cache.
+
+This value is `false` by default.
+
+### encryption_key
+
+The string used to generate a key for encrypting the cache. In order for encrypted cache data to persist between Grafana restarts, the key must be specified. If left empty when encryption is enabled, the key will be automatically generated on startup and the cache will be cleared upon restarts.
```
Comment (achatterjee-grafana) [inline docs/sources/enterprise/enterprise-configuration.md:373]:
```suggestion
This string generates a key for encrypting the cache. For the encrypted cache data to persist between Grafana restarts, this key must be specified. If it is left empty when encryption is enabled, then the key is automatically generated on startup and the cache is cleared upon restarts.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added some copy-edit suggestion.

File: docs/sources/enterprise/enterprise-configuration.md (hunk 1 above)
Comment (stephaniehingtgen) [reply docs/sources/enterprise/enterprise-configuration.md]: Updated! Sorry for the delayed update - I was out all last week.

**Review by kminehart — APPROVED** [review]

File: docs/sources/enterprise/enterprise-configuration.md [hunk 2]
```diff
@@ -361,6 +361,19 @@ This value limits the size of a single cache value. If a cache value (or query r
 
 The default is `1`.
 
+## [caching.encryption]
+### enabled
+
+Setting 'enabled' to `true` allows users to encrypt the query values in the cache.
```
Comment (achatterjee-grafana) [inline docs/sources/enterprise/enterprise-configuration.md:367]:
```suggestion
When 'enabled' to `true`, this setting allows users to encrypt the query values in the cache. Default is `false`.
```

File: docs/sources/enterprise/enterprise-configuration.md [hunk 3]
```diff
@@ -361,6 +361,19 @@ This value limits the size of a single cache value. If a cache value (or query r
...
+
+This value is `false` by default.
```
Comment (achatterjee-grafana) [inline docs/sources/enterprise/enterprise-configuration.md:369]: remove

File: docs/sources/enterprise/enterprise-configuration.md [hunk 4]
```diff
@@ -361,6 +361,19 @@ This value limits the size of a single cache value. If a cache value (or query r
...
+
+### encryption_key
+
+This string generates a key for encrypting the cache. For the encrypted cache data to persist between Grafana restarts, this key must be specified. If it is left empty when encryption is enabled, then the key is automatically generated on startup and the cache is cleared upon restarts.
```
Comment (achatterjee-grafana) [inline docs/sources/enterprise/enterprise-configuration.md:373]:
```suggestion
A string to generate a key for encrypting the cache. For the encrypted cache data to persist between Grafana restarts, you must specify this key. If it is empty when encryption is enabled, then the key is automatically generated on startup, and the cache clears upon restarts.
```

**Review by achatterjee-grafana — COMMENTED** [review]
Added doc review.

File: docs/sources/enterprise/enterprise-configuration.md (hunk 3 above)
Comment (stephaniehingtgen) [reply docs/sources/enterprise/enterprise-configuration.md]: I updated it to be "The default is `false`.", but kept it on a different line to follow the pattern of the other settings in this markdown. Does that seem okay?

File: docs/sources/enterprise/enterprise-configuration.md (hunk 2 above)
Comment (stephaniehingtgen) [reply docs/sources/enterprise/enterprise-configuration.md]: I updated it to
```
When 'enabled' is `true`, query values in the cache are encrypted.

The default is `false`.
```

Hopefully this is a bit more clear, but let me know what you think

**Comment by stephaniehingtgen** [comment]
I don't have permissions to Merge this pull request, would someone be able to merge it in?

**Comment by stephaniehingtgen** [comment]
@achatterjee-grafana are you good with this being merged in? If so, would you merge it in? These configurations are in v8.1, so it would be good to get this before the code freeze for 8.1.

**Review by achatterjee-grafana — APPROVED** [review]

**Comment by achatterjee-grafana** [comment]
Added the backport label.

**Pull request merged**